src/tracks/playabletrack/wavetrack/ui/SpectrumView.h

```diff
@@ -70,33 +76,24 @@ class SpectralData{
    }
 
    // The double time points is quantized into long long
-   void addTimeFreqData(long long ll_sc, wxInt64 freq){
+   void addHopBinData(int hopNum, int freqBin){
       // Update the start and end sampleCount of current selection
-      if(ll_sc > mEndT)
-         mEndT = ll_sc;
-      if(ll_sc < mStartT)
-         mStartT = ll_sc;
-
-      // "Time-axis rounding"
-      // Using int division to round the sampleCount to the hop size of FFT to save computation
-      int hopSize = 128;
-//      ll_sc = ll_sc / hopSize * hopSize;
-
-      // "Frequency-bin rounding"
-      // Round the exact frequency to the nearest bin
-      int binNum = freq / (mSampleRate / mWindowSize);
-
-      if(dataBuffer.find(ll_sc) == dataBuffer.end())
-         dataBuffer[ll_sc] = std::set<int>{ binNum };
+      if(hopNum * mHopSize > mEndT)
+         mEndT = hopNum * mHopSize;
+      if(hopNum * mHopSize < mStartT)
+         mStartT = hopNum * mHopSize;
+
+      if(dataBuffer.find(hopNum) == dataBuffer.end())
+         dataBuffer[hopNum] = std::set<int>{ freqBin };
       else
-         dataBuffer[ll_sc].insert(binNum);
+         dataBuffer[hopNum].insert(freqBin);
```

Comment:
You can simplify this if-else block just to the one statement after the else.

That is because operator [] of std::map may mutate the map.  If the data type has a default constructor, then it guarantees the subscripted element is present, creating it on demand in default-constructed state if it was not.